### ISSUE TITLE
backend company-lsp error "Wrong type argument: arrayp, nil" with args (prefix)

### ISSUE THREAD

**Issue by tinloaf** [tigersoldier/company-lsp#73]
I'm using `cquery` with `company-lsp`. Since the latest update, completion has stopped working. Whenever something should be completed, I'm getting this error in my messages buffer:

> Company: An error occurred in auto-begin
> Company: backend company-lsp error "Wrong type argument: arrayp, nil" with args (prefix)

This is the backtrace in the debugger when this happens:

```
Debugger entered--Lisp error: (wrong-type-argument arrayp nil)
  company-lsp(prefix)
  apply(company-lsp prefix)
  company-call-backend-raw(prefix)
  apply(company-call-backend-raw prefix)
  company--force-sync(company-call-backend-raw (prefix) company-lsp)
  company-call-backend(prefix)
  company--begin-new()
  company--perform()
  company-auto-begin()
  company-idle-begin(#<buffer edgeinsertion.cpp> #<window 3 on edgeinsertion.cpp> 49081 30939)
  apply(company-idle-begin (#<buffer edgeinsertion.cpp> #<window 3 on edgeinsertion.cpp> 49081 30939))
  timer-event-handler([t 23569 534 400764 nil company-idle-begin (#<buffer edgeinsertion.cpp> #<window 3 on edgeinsertion.cpp> 49081 30939) nil 512000])
```
I'm not sure whether this is the same as #51 and #31 - however, both these issues somehow seem garbled (are they missing messages in between? Somehow I can't quite follow the conversation.

Here are the excerpts of my config that are relevant (I think…):

```elisp
(unless (package-installed-p 'company-lsp)
  (package-install 'company-lsp))
(unless (package-installed-p 'cquery)
  (package-install 'cquery))
(unless (package-installed-p 'lsp-mode)
  (package-install 'lsp-mode))

…

;; Company Mode
(add-hook 'c++-mode-hook 'company-mode)
(add-hook 'c-mode-hook 'company-mode)
(setq company-lsp-enable-snippet 't)

;; LSP UI
(require 'lsp-ui)
(add-hook 'lsp-mode-hook 'lsp-ui-mode)
(add-hook 'lsp-ui-mode-hook
					(lambda ()
					(setq-local flycheck-checker 'c/c++-clang)))

;; Cquery
(require 'cquery)
(setq cquery-executable "/home/lukas/opt/cquery/bin/cquery")
(setq cquery-extra-init-params '(:index (:comments 2) :cacheFormat "msgpack"))
(defun cquery//enable ()
  (condition-case nil
			;;      (lsp-cquery-enable)
			(lsp)
    (user-error nil)))

(add-hook 'c-mode-hook #'cquery//enable)
(add-hook 'c++-mode-hook #'cquery//enable)

(eval-after-load 'company
  '(add-to-list 'company-backends 'company-lsp))
(setq company-transformers nil company-lsp-async t company-lsp-cache-candidates nil)

;;; Semantic highlighting
(setq cquery-sem-highlight-method 'font-lock)
(cquery-use-default-rainbow-sem-highlight)
```

Finally, some of the things that `lsp-print-io = t` gives me immediately prior to the error can be found at https://gist.github.com/tinloaf/86bc8e330e283d0b8e6e9f53a2dc9b8d

**Comment by lshoravi** [comment]
In a similiar vein my Company-lsp gives me this error with ccls:

`Company: backend company-lsp error "Wrong type argument: lsp--workspace, nil" with args (prefix)
`

using the config

```
  (use-package lsp-mode
    :commands lsp
    :config (require 'lsp-clients))

  (use-package lsp-ui
    :ensure t)

  (use-package company-lsp
    :ensure t
    :custom (push 'company-lsp company-backends))

  (use-package ccls
    :ensure t
    :custom  (setq ccls-executable "/home/linus/.emacs.d/ccls/ccls/Release/ccls")
    :hook ((c-mode c++-mode objc-mode) .
           (lambda () (progn (require 'ccls) (lsp)))))
```

**Comment by tinloaf** [comment]
Indeed - reinstalling solved it for me, too, and I also have a separate issue now. :laughing: 

I now have the error that `incf` is an unknown function, which is fixed by PR #74 .

**Comment by lshoravi** [comment]
Indeed it is! Thank you, I had the same issue.
